jetty-http2/http2-http-client-transport/src/main/java/org/eclipse/jetty/http2/client/http/HttpConnectionOverHTTP2.java

```diff
@@ -57,25 +61,35 @@ public Session getSession()
     @Override
     protected SendFailure send(HttpExchange exchange)
     {
-        exchange.getRequest().version(HttpVersion.HTTP_2);
-        normalizeRequest(exchange.getRequest());
+        HttpRequest request = exchange.getRequest();
+        request.version(HttpVersion.HTTP_2);
+        normalizeRequest(request);
 
         // One connection maps to N channels, so for each exchange we create a new channel.
-        HttpChannel channel = newHttpChannel(false);
-        channels.add(channel);
+        HttpChannelOverHTTP2 channel = provideHttpChannel();
+        activeChannels.add(channel);
 
         return send(channel, exchange);
     }
 
-    protected HttpChannelOverHTTP2 newHttpChannel(boolean push)
+    protected HttpChannelOverHTTP2 provideHttpChannel()
     {
-        return new HttpChannelOverHTTP2(getHttpDestination(), this, getSession(), push);
+        HttpChannelOverHTTP2 channel = idleChannels.poll();
+        if (channel == null)
+            channel = new HttpChannelOverHTTP2(getHttpDestination(), this, getSession());
+        return channel;
     }
 
-    protected void release(HttpChannel channel)
+    protected void release(HttpChannelOverHTTP2 channel)
```

Comment:
I kind of prefer either:  
1. This method is called release and it calls reset() on the channel etc. and adds it t o the idle pool if it is not failed
2. This method is called recycle. reset() is called elsewhere and recycle is only called if the channel is not failed